src/watch/index.ts

```diff
@@ -208,29 +234,39 @@ export const watchCommand = command({
 	 * As an alternative, we watch cwd and all run-time dependencies
 	 */
 	const watcher = watch(
-		[
-			...argv._,
-			...options.include,
-		],
+		argv._,
 		{
 			cwd: process.cwd(),
 			ignoreInitial: true,
-			ignored: [
-				// Hidden directories like .git
-				'**/.*/**',
-
-				// Hidden files (e.g. logs or temp files)
-				'**/.*',
-
-				// 3rd party packages
-				'**/{node_modules,bower_components,vendor}/**',
-
-				...options.exclude,
-			],
 			ignorePermissionErrors: true,
+			// ignore all files that are by default ignored or explicitly excluded
+			ignored: file => isDefaultIgnore(file) || isOptionsExclude(file),
 		},
 	).on('all', reRun);
 
+	if (resolvedIncludes.length > 0) {
+		const globParents = resolvedIncludes.map(pattern => (
+			isGlob(pattern)
+				? globParent(pattern)
+				: pattern
+		));
+
+		watch(globParents, {
+			cwd: process.cwd(),
+			ignoreInitial: true,
+			ignorePermissionErrors: true,
+			// ignore all files not in includes or explicitly excluded
+			// we need to make sure not to ignore directories otherwise chokidar won't check for it
+			ignored: file => (
+				!globParents.includes(file)
+				&& (
+					!isOptionsInclude(file)
+					|| isOptionsExclude(file)
+				)
+			),
+		}).on('all', reRun);
```

Comment:
Is it possible to add to the same watcher above instead of making a new one?

Reply:
I created a separate one since the logic of the ignore is different for explicit includes compared to default includes (we want to include things that might be by default excluded e.g. node_modules) and I think under the hood the chokidar would just create separate watchers for the different glob parents resulting in no performance impact.